test/integration/job/job_test.go

```diff
@@ -2117,6 +2139,140 @@ func BenchmarkLargeIndexedJob(b *testing.B) {
+			b.ResetTimer()
+			for n := 0; n < b.N; n++ {
+				b.StopTimer()
+				jobObj, err := createJobWithDefaults(ctx, clientSet, ns.Name, &batchv1.Job{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: fmt.Sprintf("npods-%d-%d-%v", tc.nPods, n, enableJobBackoffLimitPerIndex),
+					},
+					Spec: batchv1.JobSpec{
+						Parallelism:          ptr.To(tc.nPods),
+						Completions:          ptr.To(tc.nPods),
+						CompletionMode:       &mode,
+						BackoffLimitPerIndex: tc.backoffLimitPerIndex,
+						BackoffLimit:         ptr.To(tc.nPods),
+					},
+				})
+				if err != nil {
+					b.Fatalf("Failed to create Job: %v", err)
+				}
+				validateJobPodsStatus(ctx, b, clientSet, jobObj, podsByStatus{
+					Active: int(tc.nPods),
+					Ready:  ptr.To[int32](0),
+				})
+
+				b.StartTimer()
+				remaining := 0
+				remaining = int(tc.nPods)
+				if err := wait.ExponentialBackoff(backoff, func() (done bool, err error) {
+					if err, fail := setJobPodsPhase(ctx, clientSet, jobObj, v1.PodFailed, remaining); err != nil {
+						remaining -= fail
+						b.Logf("Transient failure failing pods: %v", err)
+						return false, nil
+					}
+					return true, nil
+				}); err != nil {
+					b.Fatalf("Could not succeed the remaining %d pods: %v", remaining, err)
+				}
+				validateJobPodsStatus(ctx, b, clientSet, jobObj, podsByStatus{
+					Active: int(tc.nPods),
+					Ready:  ptr.To[int32](0),
+					Failed: int(tc.nPods),
+				})
+
+				// Cleanup Pods and Job.
+				b.StopTimer()
+
+				// make the job succeeded
+				remaining = int(tc.nPods)
+				if err := wait.ExponentialBackoff(backoff, func() (done bool, err error) {
+					if err, succ := setJobPodsPhase(ctx, clientSet, jobObj, v1.PodSucceeded, remaining); err != nil {
+						remaining -= succ
+						b.Logf("Transient failure succeeding pods: %v", err)
+						return false, nil
+					}
+					return true, nil
+				}); err != nil {
+					b.Fatalf("Could not succeed the remaining %d pods: %v", remaining, err)
+				}
+				validateJobPodsStatus(ctx, b, clientSet, jobObj, podsByStatus{
+					Ready:     ptr.To[int32](0),
+					Succeeded: int(tc.nPods),
+					Failed:    int(tc.nPods),
+				})
+				validateJobSucceeded(ctx, b, clientSet, jobObj)
+
+				// Clean up pods in pages, because DeleteCollection might timeout.
+				// #90743
+				for {
+					pods, err := clientSet.CoreV1().Pods(ns.Name).List(ctx, metav1.ListOptions{Limit: 1})
+					if err != nil {
+						b.Fatalf("Failed to list Pods for cleanup: %v", err)
+					}
+					if len(pods.Items) == 0 {
+						break
+					}
+					err = clientSet.CoreV1().Pods(ns.Name).DeleteCollection(ctx,
+						metav1.DeleteOptions{},
+						metav1.ListOptions{
+							Limit: 1000,
+						})
+					if err != nil {
+						b.Fatalf("Failed to cleanup Pods: %v", err)
+					}
+				}
+				err = clientSet.BatchV1().Jobs(jobObj.Namespace).Delete(ctx, jobObj.Name, metav1.DeleteOptions{})
+				if err != nil {
+					b.Fatalf("Failed to cleanup Job: %v", err)
+				}
```

Comment:
Let's put this in a function for both test to use and call with `b.CleanUp`

Reply:
Done